packages/@aws-cdk/custom-resource-handlers/lib/aws-route53/cross-account-zone-delegation-handler/index.ts

```diff
@@ -77,7 +77,12 @@ async function cfnEventHandler(props: ResourceProperties, isDeleteEvent: boolean
 
 async function getHostedZoneIdByName(name: string, route53: Route53): Promise<string> {
   const zones = await route53.listHostedZonesByName({ DNSName: name });
-  const matchedZones = zones.HostedZones?.filter(zone => zone.Name === `${name}.`) ?? [];
+  const matchedZones = zones.HostedZones?.filter(zone => {
+    const matchZoneName = zone.Name === `${name}.`;
+    const isPublic = zone.Config?.PrivateZone === false || zone.Config?.PrivateZone === undefined;
```

Comment:
does `const isPublic = zone.Config?.PrivateZone !== true` work here?

Reply:
Yea. Should work and it makes the logic simpler. Will update